src/EditorFeatures/Core.Wpf/InlineDiagnostics/InlineDiagnosticsTaggerProvider.cs

```diff
@@ -62,6 +64,20 @@ protected internal override bool IncludeDiagnostic(DiagnosticData diagnostic)
                 !diagnostic.IsSuppressed;
         }
 
+        protected internal override bool IsEnabled(Document document)
+        {
+            var workspace = document.Project.Solution.Workspace;
+            var option = workspace.Options.GetOption(InlineDiagnosticsOptions.EnableInlineDiagnostics, document.Project.Language);
+            if (_experimentEnabled is null)
+            {
+                var experimentationService = document.Project.Solution.Workspace.Services.GetRequiredService<IExperimentationService>();
+                _experimentEnabled = experimentationService.IsExperimentEnabled(WellKnownExperimentNames.InlineDiagnostics);
```

Comment:
i don't get why you need to cache this value.

Reply:
Would I want to get the experimentation service every time produced tags is called? This way it doesn't need to call it every time right?